### PR TITLE
[G2M] Skeleton Layout

### PR DESCRIPTION
**Pull request by Hamaad102**
Adds Skeleton Layout

- Added the Skeleton component
- Takes an optional prop (view) that in future iterations will allow the user to select different skeleton wireframes
- Current implementation is just an article

https://user-images.githubusercontent.com/18343242/204023790-3e9ab131-f738-47bd-a56f-f2277d3585cf.mov

### CODE REVIEW

File: src/components/skeleton/article.js
Comment (geoerika) [inline src/components/skeleton/article.js:47]: Need default value for height as well.

Comment (geoerika) [inline src/components/skeleton/article.js:47]: `customSkeleton` is object but default is string.

File: src/components/skeleton/article.js
Comment (geoerika) [inline src/components/skeleton/article.js:37]: Perhaps be more specific with the types here:
```
skeleton: PropTypes.shape({
  background: PropTypes.string,
  ...
```

File: src/components/skeleton/article.js
Comment (geoerika) [inline src/components/skeleton/article.js:21]: We should add description for this , I am guessing height is in pixels?

File: src/components/skeleton/index.js
Comment (geoerika) [inline src/components/skeleton/index.js:7]: Two empty lines between imports and rest of code block please.

File: src/components/skeleton/index.js
Comment (geoerika) [inline src/components/skeleton/index.js:62]: No need for 2 empty lines here, keep it everywhere 1 empty line, except after imports.

Comment (geoerika) [reply src/components/skeleton/index.js:62]: Perhaps add this rule to lint:
```
'import/newline-after-import': ['error', { 'count': 2 }],
```

File: src/components/skeleton/index.js
Comment (geoerika) [inline src/components/skeleton/index.js:96]: I think it is safe to give the default value as `''` instead of `null`, no?

File: src/components/skeleton/index.js
Comment (geoerika) [inline src/components/skeleton/index.js:91]: See above for `skeleton ` comment in the other file.

**Review by geoerika — APPROVED** [review]
Looks good! Just minor changes required.

File: src/components/skeleton/article.js
Comment (kc-leung) [inline src/components/skeleton/article.js:32]: we have to add container name for at least the main div see other components for example. This will help to debug and for custom styling for the user

Comment (kc-leung) [inline src/components/skeleton/article.js:32]: we have to add container name for at least the main div see other components for example. This will help to debug and for custom styling for the user

File: src/components/skeleton/index.js
Comment (kc-leung) [inline src/components/skeleton/index.js:86]: same with above regarding the div container className

File: src/components/skeleton/article.js
Comment (kc-leung) [inline src/components/skeleton/article.js:15]: lets try to use as much of existing classes from tailwind if it already exits in tailwind config lets use them. Only do it in makestyle way for complicated cases or child inner container. see existing components for examples. https://tailwindcss.com/docs/width classes from tailwind also you can take look in tailwindConfig.js

File: src/components/skeleton/index.js
Comment (kc-leung) [inline src/components/skeleton/index.js:61]: same with above regarding the tailwind existing classes

File: src/components/skeleton/index.js
Comment (kc-leung) [inline src/components/skeleton/index.js:64]: please, add forwardRef to allow user to select it since the components in lumen-labs are all re-usable components. See existing components for example.

**Review by kc-leung — CHANGES_REQUESTED** [review]

**Review by kc-leung — APPROVED** [review]
looks nice, lets wait for @vlatawiec review before merging

**Review by vlatawiec — APPROVED** [review]
Looks good for v1! Thanks for your hard work!

**Pull request merged**
